tests/test_prepend.py

```diff
@@ -0,0 +1,120 @@
+
+    def test_PrependEffect(self):
+        """Check that a concatenated FrameSet transforms correctly.
+        """
+        set1 = self.makeFrameSet(2, 3)
+        set2 = self.makeFrameSet(3, 1)
+        set12 = prepend(set2, set1)
+
+        # Can't match 1D output to 2D input
+        with self.assertRaises(RuntimeError):
+            prepend(set1, set2)
+
+        x = [1.2, 3.4]
+        y_merged = set12.tranForward(x)
+        y_separate = set2.tranForward(set1.tranForward(x))
+        assert_allclose(y_merged, y_separate)
+
+        y = [-0.3]
+        x_merged = set12.tranInverse(y)
+        x_separate = set1.tranInverse(set2.tranInverse(y))
+        assert_allclose(x_merged, x_separate)
+
+    def test_PrependFrames(self):
+        """Check that a concatenated FrameSet preserves all Frames.
+        """
+        set1 = self.makeFrameSet(1, 3)
+        # AST docs say FrameSets always have contiguously numbered frames,
+        # but let's make sure
+        set1.removeFrame(2)
+        set2 = self.makeFrameSet(3, 2)
+        set12 = prepend(set2, set1)
+
+        self.assertEquals(set1.getNframe() + set2.getNframe(),
+                          set12.getNframe())
+        for i in range(1, 1+set1.getNframe()):
+            oldFrame = set1.getFrame(i)
+            newFrame = set12.getFrame(i)
+            self.assertEquals(oldFrame.getIdent(), newFrame.getIdent())
+            self.assertEquals(oldFrame.getNaxes(), newFrame.getNaxes())
+            if i == set1.getBase():
+                self.assertTrue(i == set12.getBase())
+            else:
+                self.assertFalse(i == set12.getBase())
+        for i in range(1, 1+set2.getNframe()):
+            offset = set1.getNframe()
+            oldFrame = set2.getFrame(i)
+            newFrame = set12.getFrame(offset + i)
+            self.assertEquals(oldFrame.getIdent(), newFrame.getIdent())
+            self.assertEquals(oldFrame.getNaxes(), newFrame.getNaxes())
+            if i == set2.getCurrent():
+                self.assertTrue(offset + i == set12.getCurrent())
+            else:
+                self.assertFalse(offset + i == set12.getCurrent())
+
+    def test_PrependIndependent(self):
+        """Check that a concatenated FrameSet is not affected by changes
+        to its constituents.
+        """
+        set1 = self.makeFrameSet(3, 3)
+        set2 = self.makeFrameSet(3, 3)
+        set12 = prepend(set2, set1)
+
+        nTotal = set12.getNframe()
+        x = [1.2, 3.4, 5.6]
+        y = set12.tranForward(x)
+
+        set1.addFrame(2, makeTwoWayPolyMap(4, 2), Frame(2, "Ident=extra"))
+        set1.addFrame(1, makeTwoWayPolyMap(3, 3), Frame(3, "Ident=legume"))
+        set1.removeFrame(3)
+        set2.addFrame(4, makeForwardPolyMap(1, 4), Frame(4, "Ident=extra"))
+        set2.setBase(2)
+
+        # Use exact equality because nothing should change
+        self.assertEquals(set12.getNframe(), nTotal)
+        self.assertEquals(set12.tranForward(x), y)
```

Comment:
Given your reply above re: compatibility requirements, should I still have a test checking that Frame(2) -> SkyFrame is legal (which it currently is)?